src/Main.elm

```diff
@@ -101,8 +106,22 @@ init location =
 type Msg
     = SelectMovie
     | MovieSelected ( Maybe Movie, List Movie )
+    | FocusMovie (Maybe Movie)
```

Comment:
If I'm understanding the rest of the code correctly, it looks like `FocusMovie (Just movie)` is being used to show a movie and `FocusMovie Nothing` is being used to close the modal.  I think the code would be easier to understand/maintain with separate events. Maybe: `PreviewMovie Movie` and `CloseMovieModal`.

Reply:
More generally speaking, the way I came to this conclusion is that I've gotten in the habit of looking for ways to avoid `Maybe` in code.  It's not that `Maybe` is _always_ bad, just that having a more explicit representation, or leveraging a default state is often clearer.